pkg/controller/iam/user/controller.go

```diff
@@ -179,3 +224,56 @@ func (e *external) Delete(ctx context.Context, mgd resource.Managed) error {
 
 	return awsclient.Wrap(resource.Ignore(iam.IsErrorNotFound, err), errDelete)
 }
+
+type tagger struct {
+	kube client.Client
+}
+
+type externalNamer struct {
+	kube client.Client
+}
+
+func (t *tagger) Initialize(ctx context.Context, mgd resource.Managed) error {
+	cr, ok := mgd.(*v1beta1.User)
+	if !ok {
+		return errors.New(errUnexpectedObject)
+	}
+
+	added := false
+	defaultTags := resource.GetExternalTags(mgd)
+
+	for i, t := range cr.Spec.ForProvider.Tags {
```

Comment:
nitpick: Implementation [here](https://github.com/crossplane/provider-aws/blob/793e588/pkg/controller/ecr/repository/controller.go#L257) you can copy seems to be doing the same thing maybe with fewer lines but both should work well.

Reply:
I had a look at the suggested code, and I found the following divergences:
- it doesn't handle tags with empty values
- it allows users to change default tags values (actually appending two Tag elements with the same key but different values to Spec.ForProvider.Tags, guessing the last one will be "winning")

is this the intended behavior?